File: Solution-Python/Max_1004.py
class Solution:
    def longestOnes(self, A: List[int], K: int) -> int:
        l = 0
        for a in A:
            if a == 0:
                K -= 1
            if K < 0:
                if A[l] == 0:
                    K += 1
                l += 1
        return len(A) - l

File: Solution-Java/Max_1004.java
class Max_1004 {
    public int longestOnes(int[] A, int K) {
        int l = 0;
        for (int a : A) {
            if (a == 0) K--;
            if (K < 0 && A[l++] == 0) K++;
        }
        return A.length - l;
    }
}